src/components/Calculator.js

```diff
@@ -1,89 +1,58 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Calculator.css';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: 0,
-    };
+const Calculator = () => {
+  const [input, setInput] = useState({
+    total: 0,
+    next: 0,
+  });
 
-    this.click = this.click.bind(this);
-  }
+  const outPut = (value) => {
+    const finalResultState = calculate(input, value);
+    setInput(finalResultState);
+  };
 
-  click(event) {
-    this.setState((obj) => calculate(obj, event.target.innerText));
-  }
+  const click = (event) => {
+    const btnTextValue = event.target.innerText;
+    outPut(btnTextValue);
+  };
 
-  render() {
-    let lblValue = 0;
-    const { total, next } = this.state;
-    if (next) {
-      lblValue = next;
-    } else if (total) {
-      lblValue = total;
-    }
+  let lblValue = 0;
 
-    return (
-      <div className="App">
-        <div className="calculator-grid">
-          <div className="output">
-            <div className="current-operand">{lblValue}</div>
-          </div>
-          <button type="button" onClick={this.click}>AC</button>
-          <button type="button" onClick={this.click}>+/-</button>
-          <button type="button" onClick={this.click}>%</button>
-          <button type="button" onClick={this.click} className="orange-buttton">&divide;</button>
-          <button type="button" onClick={this.click}>7</button>
-          <button type="button" onClick={this.click}>8</button>
-          <button type="button" onClick={this.click}>9</button>
-          <button type="button" onClick={this.click} className="orange-buttton">x</button>
-          <button type="button" onClick={this.click}>4</button>
-          <button type="button" onClick={this.click}>5</button>
-          <button type="button" onClick={this.click}>6</button>
-          <button type="button" onClick={this.click} className="orange-buttton">-</button>
-          <button type="button" onClick={this.click}>1</button>
-          <button type="button" onClick={this.click}>2</button>
-          <button type="button" onClick={this.click}>3</button>
-          <button type="button" onClick={this.click} className="orange-buttton">+</button>
-          <button type="button" onClick={this.click} className="span-two">0</button>
-          <button type="button" onClick={this.click}>.</button>
-          <button type="button" onClick={this.click} className="orange-buttton">=</button>
-        </div>
-      </div>
-    );
+  if (input.next) {
+    lblValue = input.next;
+  } else if (input.total) {
+    lblValue = input.total;
   }
```

Comment:
[OPTIONAL] I think you could improve the user experience by displaying the operation the user has chosen as well. This way the user will always know at what point of an operation they are